mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -4308,6 +4309,13 @@ func containerClusterPrivateClusterConfigCustomDiff(_ context.Context, d *schema
 	}
 	config := pccList[0].(map[string]interface{})
 	if config["enable_private_nodes"].(bool) {
+		// Skip checking the existence of master ipv4 cidr if Autopilot is enabled.
+		// Master ipv4 cidr is optional for Autopilot.
+		// https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#auto_subnet
+		if d.Get("enable_autopilot").(bool) {
+			return nil
+		}
```

Comment:
It seems Terraform [no longer rejects](https://storage.cloud.google.com/vcr-test-logs/beta/refs/heads/auto-pr-5979/artifacts/508889ff-9a07-426f-8bd3-a539c1112a56/recording/TestAccContainerCluster_withPrivateClusterConfigMissingCidrBlock.log) missing master_ipv4_cidr_block for non-Autopilot clusters after I set `Computed: true`. @c2thorn Could you help?

Reply:
This is a tough situation, as `master_ipv4_cidr_block` is `Computed` only on the condition that `enable_autopilot` is true, but there is not a representation for that in Terraform. Setting `Computed` is the only way we can get Terraform to accept the API-generated value for the autopilot scenario.

The current validation has a workaround for https://github.com/hashicorp/terraform-provider-google/issues/4186 that relies on the value being known, and it will never be known when `Computed`.

I'm not 100% sure about the solution yet. Options I can come up with are to remove the validation workaround (breaking change), or move the validation to right before create/update when we know exactly what value we are about to send. Apologies for the wait, but I need some time to ask the team if they have any other ideas.

Reply:
no worries. Thanks @c2thorn. Yeah I was also reading https://github.com/hashicorp/terraform-provider-google/issues/4186#issuecomment-536118762 this afternoon :) 

agreed it's challenging. Removing the workaround doesn't sound viable since it will break https://github.com/hashicorp/terraform-provider-google/issues/4186 again. Looking for alternatives.

Reply:
another option might be removing the validation and delegating the validation to GKE API